browser/src/slideshow/SlideShowPresenter.ts

```diff
@@ -247,12 +247,34 @@ class SlideShowPresenter {
 	}
 
 	_doPresentation() {
-		const blankTexture = this._slideRenderer.createEmptyTexture(
-			this._presentationInfo.docWidth,
-			this._presentationInfo.docHeight,
-		);
+		const slideInfo = this.getSlideInfo(this._currentSlide);
+
+		// To speed up the process, if we have transition info, then only render
+		// a black empty slide as the first slide. otherwise, directly render the first slide.
+		if (
+			slideInfo?.transitionType != undefined &&
```

Comment:
@quikee Should we add more checks here for animation/video?

Reply:
I don't see the need..